### PR TITLE
Added JWT support. Implemented login logic.

### PR DESCRIPTION
**Pull request by Vadym-Ozi**

### CODE REVIEW

File: src/main/java/example/Application.java [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.PropertySource;
 
 @SpringBootApplication
+//@PropertySource("classpath:application.properties")
```
Comment (Elena-Bruyako) [inline src/main/java/example/Application.java:8]: remove comment

File: src/main/java/example/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/example/security/JwtAuthenticationFilter.java:43]: Use Spring constant AUTHORIZATION from HttpHeaders

File: src/main/java/example/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/example/security/JwtAuthenticationFilter.java:44]: Create constant for "Bearer "

File: src/main/java/example/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/example/security/JwtAuthenticationFilter.java:45]: use constant.length()

File: src/test/java/example/ApplicationTests.java [hunk 5]
```diff
@@ -6,8 +6,9 @@
 @SpringBootTest
 class ApplicationTests {
 
-    @Test
-    void contextLoads() {
-    }
+    // Temporary solution until i learn SBoot Test
```
Comment (Elena-Bruyako) [inline src/test/java/example/ApplicationTests.java:9]: remove commented code

File: src/main/java/example/controller/AuthenticationController.java [hunk 6]
```diff
@@ -18,9 +21,15 @@
...
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto registration(@RequestBody @Valid UserRegistrationRequestDto request) throws RegistrationException {
         return userService.register(request);
     }
+
+    @PostMapping("/login")
```
Comment (Elena-Bruyako) [inline src/main/java/example/controller/AuthenticationController.java:31]: Add swagger annotations for endpoints

File: src/main/java/example/security/JwtUtil.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                .setSubject(username)
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + expiration))
+                .signWith(secret)
+                .compact();
+    }
+
+    public boolean isValidToken(String token) {
+        try
+        {
```
Comment (Elena-Bruyako) [inline src/main/java/example/security/JwtUtil.java:36]:
```suggestion
        try {
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/example/security/JwtAuthenticationFilter.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import java.io.IOException;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private final JwtUtil jwtUtil;
+    private final UserDetailsService userDetailsService;
+    private static final String HEADER = "Authorization";
+    private static final String TOKEN_NAME = "Bearer ";
+    private static final int SUBSTRING_INDEX = 7;
```
Comment (okuzan) [inline src/main/java/example/security/JwtAuthenticationFilter.java:25]: just use TOKEN_NAME.length()

Comment (Vadym-Ozi) [reply src/main/java/example/security/JwtAuthenticationFilter.java:25]: just now realized

**Pull request merged**